### PR TITLE
Force the user to specify the partition data type for external tables…

### PR DESCRIPTION
**Pull request by jfilali**
… and force to use a scalar data type

### CODE REVIEW

**Comment by dgomezferro** [comment]
Do we have a list somewhere of the supported types for partitioned columns? Why are blobs OK but dates aren't?

Also, please tag the commit with the JIRA number so it's easier to identify looking at the log

File: db-tools-i18n/src/main/resources/com/splicemachine/db/loc/messages.xml [hunk 1]
```diff
@@ -8964,6 +8964,19 @@ Shutting down instance {0} on database directory {1} with class loader {2} </tex
...
 
+           <msg>
+               <name>EXT21</name>
+               <text>'{0} is not defined in the table definition'</text>
+               <arg>columnName</arg>
+           </msg>
+
+           <msg>
+               <name>EXT22</name>
+               <text>'Partition column {0} need to be numerical '</text>
```
Comment (scottfines) [inline db-tools-i18n/src/main/resources/com/splicemachine/db/loc/messages.xml]: 'numerical' isn't the same thing as 'scalar'; a float is numerical but is not a scalar.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj [hunk 2]
```diff
@@ -11749,11 +11750,25 @@ tableDefinition() throws StandardException :
...
+                if (isExternal && partitionedResultColumns != null){
+                     List<String> columnsPartitions = Arrays.asList(partitionedResultColumns.getColumnNames());
+
+                     for(String columnPartititon : columnsPartitions){
+                        ColumnDefinitionNode definition = tableElementList.findColumnDefinition(columnPartititon);
+                        if(definition ==  null){
+                                throw StandardException.newException(SQLState.EXTERNAL_TABLES_PARTITIONS_REQUIRED, columnPartititon);
+                            }
+                        if(!definition.getType().getTypeId().isScalar()){
+                                throw StandardException.newException(SQLState.EXTERNAL_TABLES_SCALAR_CONSTRAINT, columnPartititon);
```
Comment (scottfines) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj]: why do we limit the column types to scalar only?

File: db-tools-i18n/src/main/resources/com/splicemachine/db/loc/messages.xml (hunk 1 above)
Comment (jfilali) [inline db-tools-i18n/src/main/resources/com/splicemachine/db/loc/messages.xml]: yes true, I need instead to test if id either numerical or scalar, there is a confusion there

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj (hunk 2 above)
Comment (jfilali) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj]: Currently, numeric data types and string type are supported in Spark

**Pull request closed without merging**